Source/Core/DolphinQt/Config/Mapping/IOWindow.cpp

```diff
@@ -270,55 +279,79 @@ void IOWindow::CreateMainLayout()
-  m_functions_combo->addItem(QStringLiteral("relative"));
-  m_functions_combo->addItem(QStringLiteral("pulse"));
-  m_functions_combo->addItem(QStringLiteral("sin"));
-  m_functions_combo->addItem(QStringLiteral("cos"));
-  m_functions_combo->addItem(QStringLiteral("tan"));
-  m_functions_combo->addItem(QStringLiteral("asin"));
-  m_functions_combo->addItem(QStringLiteral("acos"));
-  m_functions_combo->addItem(QStringLiteral("atan"));
-  m_functions_combo->addItem(QStringLiteral("atan2"));
-  m_functions_combo->addItem(QStringLiteral("sqrt"));
-  m_functions_combo->addItem(QStringLiteral("pow"));
-  m_functions_combo->addItem(QStringLiteral("min"));
-  m_functions_combo->addItem(QStringLiteral("max"));
-  m_functions_combo->addItem(QStringLiteral("clamp"));
+  // This might an empty unselectable whitespace at the end of the ComboBox selection
+  m_functions_combo->insertSeparator(m_functions_combo->count());  // A separator is also an item
+  m_functions_parameters.push_back(QStringLiteral(""));            // Keep the indexes aligned
+  // Logic/Math:
+  AddFunction("if");
+  AddFunction("not");
+  AddFunction("min");
+  AddFunction("max");
+  AddFunction("clamp");
+  AddFunction("minus");
+  AddFunction("pow");
+  AddFunction("sqrt");
+  AddFunction("sin");
+  AddFunction("cos");
+  AddFunction("tan");
+  AddFunction("asin");
+  AddFunction("acos");
+  AddFunction("atan");
+  AddFunction("atan2");
+  m_functions_combo->insertSeparator(m_functions_combo->count());
+  m_functions_parameters.push_back(QStringLiteral(""));
+  // State/time based:
+  AddFunction("onPress");
+  AddFunction("onRelease");
+  AddFunction("onChange");
+  AddFunction("cache");
+  AddFunction("hold");
+  AddFunction("toggle");
+  AddFunction("tap");
+  AddFunction("relative");
+  AddFunction("smooth");
+  AddFunction("pulse");
+  AddFunction("timer");
+  m_functions_combo->insertSeparator(m_functions_combo->count());
+  m_functions_parameters.push_back(QStringLiteral(""));
+  // Stick helpers:
+  AddFunction("deadzone");
+  AddFunction("antiDeadzone");
+  AddFunction("bezierCurve");
+  AddFunction("antiAcceleration");
 
   // Devices
   m_main_layout->addWidget(m_devices_combo);
 
   // Range
   auto* range_hbox = new QHBoxLayout();
-  range_hbox->addWidget(new QLabel(tr("Range")));
+  range_hbox->addWidget(new QLabel(tr("Range (%)")));
   range_hbox->addWidget(m_range_slider);
   range_hbox->addWidget(m_range_spinbox);
   m_range_slider->setMinimum(-500);
   m_range_slider->setMaximum(500);
   m_range_spinbox->setMinimum(-500);
   m_range_spinbox->setMaximum(500);
-  m_main_layout->addLayout(range_hbox);
+  m_range_slider->setToolTip(
+      tr("Multiplies (not clamp) the final result. Usually defaults to 100"));
+  m_range_spinbox->setToolTip(m_range_slider->toolTip());
+  // If this is an input setting, the range won't be saved in our config,
+  // nor it would make sense to change it, so just hide it
+  if (m_type != Type::InputSetting)
```

Comment:
Unfortunately the (poorly named) "Range" setting IS needed for outputs. Steering wheel forces (which are sent as values from -1 to +1) often need reversing using a negative range.

Reply:
Range is still used for outputs, this is for input settings, which is a sub category of inputs with a fixed value. They can't use range as it's never read nor serealized.

Reply:
Ah. I misread what this was for. :P